database/src/main/kotlin/no/nav/su/se/bakover/database/grunnlag/VilkårsvurderingPostgresRepo.kt

```diff
@@ -0,0 +1,142 @@
+import no.nav.su.se.bakover.database.Session
+import no.nav.su.se.bakover.database.hentListe
+import no.nav.su.se.bakover.database.oppdatering
+import no.nav.su.se.bakover.database.tidspunkt
+import no.nav.su.se.bakover.database.uuid
+import no.nav.su.se.bakover.database.uuidOrNull
+import no.nav.su.se.bakover.database.withSession
+import no.nav.su.se.bakover.database.withTransaction
+import no.nav.su.se.bakover.domain.grunnlag.Grunnlag
+import no.nav.su.se.bakover.domain.vilkår.Resultat
+import no.nav.su.se.bakover.domain.vilkår.Vilkår
+import no.nav.su.se.bakover.domain.vilkår.Vurderingsperiode
+import java.util.UUID
+import javax.sql.DataSource
+
+internal class VilkårsvurderingPostgresRepo(
+    private val dataSource: DataSource,
+    private val grunnlagRepo: GrunnlagRepo,
+) : VilkårsvurderingRepo {
+
+    override fun lagre(behandlingId: UUID, vilkår: Vilkår<Grunnlag.Uføregrunnlag>) {
+        dataSource.withTransaction { tx ->
+            slettForBehandlingId(behandlingId, tx)
+            when (vilkår) {
+                Vilkår.IkkeVurdert.Uførhet -> Unit
+                is Vilkår.Vurdert.Uførhet -> {
+                    grunnlagRepo.lagre(behandlingId, vilkår.grunnlag, tx)
+                    vilkår.vurderingsperioder.forEach {
+                        lagre(behandlingId, it, tx)
+                    }
+                }
+            }
+        }
+    }
+
+    private fun lagre(behandlingId: UUID, vurderingsperiode: Vurderingsperiode<Grunnlag.Uføregrunnlag>, session: Session) {
+        """
+                insert into vilkårsvurdering_uføre
+                (
+                    id,
+                    opprettet,
+                    behandlingId,
+                    uføre_grunnlag_id,
+                    vurdering,
+                    resultat,
+                    begrunnelse,
+                    fraOgMed,
+                    tilOgMed
+                ) values 
+                (
+                    :id,
+                    :opprettet,
+                    :behandlingId,
+                    :ufore_grunnlag_id,
+                    :vurdering,
+                    :resultat,
+                    :begrunnelse,
+                    :fraOgMed,
+                    :tilOgMed
+                )
+        """.trimIndent()
+            .oppdatering(
+                mapOf(
+                    "id" to vurderingsperiode.id,
+                    "opprettet" to vurderingsperiode.opprettet,
+                    "behandlingId" to behandlingId,
+                    "ufore_grunnlag_id" to vurderingsperiode.grunnlag?.id,
+                    "vurdering" to when (vurderingsperiode) {
+                        is Vurderingsperiode.Manuell -> "MANUELL"
+                    },
+                    "resultat" to when (vurderingsperiode.resultat) {
+                        Resultat.Avslag -> "AVSLAG"
+                        Resultat.Innvilget -> "INNVILGET"
+                        Resultat.Uavklart -> "UAVKLART"
+                    },
```

Comment:
Kunne det gått an å bruke f.eks. en enum for mapping her, evt. samle det i en funksjon ellerno? Føles litt skummelt å skulle ha noen "random strenger" gjemt inni spørringen her. I det minste legge det sånn at det er nærmere konverteringa som er andre veien?

Reply:
Ja. Noen ting her har ikke fått veldig mye kjærlighet siden iterasjon 1.